skactiveml/pool/regression/_greedy_sampling.py

```diff
@@ -0,0 +1,342 @@
+            If true, also return the utilities based on the query strategy.
+
+        Returns
+        -------
+        query_indices : numpy.ndarray of shape (batch_size)
+            The query_indices indicate for which candidate sample a label is
+            to queried, e.g., `query_indices[0]` indicates the first selected
+            sample.
+            If candidates is None or of shape (n_candidates), the indexing
+            refers to samples in X.
+            If candidates is of shape (n_candidates, n_features), the indexing
+            refers to samples in candidates.
+        utilities : numpy.ndarray of shape (batch_size, n_samples) or
+            numpy.ndarray of shape (batch_size, n_candidates)
+            The utilities of samples after each selected sample of the batch,
+            e.g., `utilities[0]` indicates the utilities used for selecting
+            the first sample (with index `query_indices[0]`) of the batch.
+            Utilities for labeled samples will be set to np.nan.
+            If candidates is None or of shape (n_candidates), the indexing
+            refers to samples in X.
+            If candidates is of shape (n_candidates, n_features), the indexing
+            refers to samples in candidates.
+        """
+
+        X, y, candidates, batch_size, return_utilities = self._validate_data(
+            X, y, candidates, batch_size, return_utilities, reset=True
+        )
+
+        X_cand, mapping = self._transform_candidates(candidates, X, y)
+
+        query_indices = np.zeros(batch_size, dtype=int)
+        sample_indices = np.arange(len(X), dtype=int)
+
+        if mapping is None:
+            X_all = np.append(X, X_cand, axis=0)
+            selected_indices = labeled_indices(y)
+            candidate_indices = len(X) + np.arange(len(X_cand), dtype=int)
+        else:
+            X_all = X
+            selected_indices = labeled_indices(y)
+            candidate_indices = mapping
+
+        utilities = np.full((batch_size, len(X_all)), np.nan)
+        distances = pairwise_distances(X_all, metric=self.metric)
+
+        for i in range(batch_size):
+            if selected_indices.shape[0] == 0:
+                dist = distances[candidate_indices][:, sample_indices]
+                util = -np.sum(dist, axis=1)
+            else:
+                dist = distances[candidate_indices][:, selected_indices]
+                util = np.min(dist, axis=1)
+            utilities[i, candidate_indices] = util
+
+            idx = rand_argmax(util, random_state=self.random_state)
+            query_indices[i] = candidate_indices[idx]
+            selected_indices = np.append(
+                selected_indices, candidate_indices[idx], axis=0
+            )
+            candidate_indices = np.delete(candidate_indices, idx, axis=0)
+
+        if mapping is None:
+            query_indices = query_indices - len(X)
+            utilities = utilities[:, len(X) :]
```

Comment:
minor change, just deleted the space in front of ':'

```suggestion
            utilities = utilities[:, len(X):]
```